DemoUI/js/app.js

```diff
@@ -0,0 +1,243 @@
+    backupDetailPage.style.zIndex = '100';
+    backupDetailPage.setAttribute('class', 'go-deeper-in');
+  }
+
+  /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+   * HISTORY HELPERS                                                               *
+   * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+  function dateFormat(date) {
+    if (date === null) {
+      return 'Never';
+    }
+
+    var minutes = date.getMinutes();
+    if (minutes < 10) {
+      minutes = '0' + minutes;
+    }
+    return date.getMonth() + '/' + date.getDate() + '/' + date.getFullYear() + ' ' +
+      date.getHours() + ':' + minutes;
+  }
+
+  function sizeFormat(size) {
+    // We'll go up to Gigabytes, but no further. If you have a terabyte SD card...
+    // I'm sorry.
+    var units = ' bytes';
+    if (size > 1024) {
+      size /= 1024;
+      units = ' MB';
+    }
+    if (size > 1024) {
+      size /= 1024;
+      units = ' GB';
+    }
+
+    return size.toFixed(2) + units;
+  }
+
+  function refreshHistories() {
+    var photoUpdatedAt = document.getElementById('photo-hist-date');
+    photoUpdatedAt.textContent = dateFormat(ffosbr.history.getValue('photos', 'lastBackupDate'));
+    var photoSize = document.getElementById('photo-hist-size');
+    photoSize.textContent = sizeFormat(ffosbr.history.getValue('photos', 'backupSize'));
+
+    var videoUpdatedAt = document.getElementById('video-hist-date');
+    videoUpdatedAt.textContent = dateFormat(ffosbr.history.getValue('videos', 'lastBackupDate'));
+    var videoSize = document.getElementById('video-hist-size');
+    videoSize.textContent = sizeFormat(ffosbr.history.getValue('videos', 'backupSize'));
+
+    var contactUpdatedAt = document.getElementById('contact-hist-date');
+    contactUpdatedAt.textContent = dateFormat(ffosbr.history.getValue('contacts', 'lastBackupDate'));
+    var contactSize = document.getElementById('contact-hist-size');
+    contactSize.textContent = sizeFormat(ffosbr.history.getValue('contacts', 'backupSize'));
+
+    var messagesUpdatedAt = document.getElementById('messages-hist-date');
+    messagesUpdatedAt.textContent = dateFormat(ffosbr.history.getValue('messages', 'lastBackupDate'));
+    var messagesSize = document.getElementById('messages-hist-size');
+    messagesSize.textContent = sizeFormat(ffosbr.history.getValue('messages', 'backupSize'));
+  }
+
+  refreshHistories();
+
+
+  /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+   * EVENT LISTENERS                                                               *
+   * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+  document.getElementById('backup-button').addEventListener('click', function() {
+    var flag = confirm('Warning! This will erase any other backup previously stored on SD card');
+    if (flag === false) {
```

Comment:
Is this where you want the code? Also should it just be if(flag ===true){} so we don't need an else?...